payjoin/src/receive/mod.rs

```diff
@@ -791,7 +808,34 @@ impl PayjoinProposal {
             self.payjoin_psbt.inputs[i].final_script_sig = None;
             self.payjoin_psbt.inputs[i].final_script_witness = None;
         }
-        Ok(self.payjoin_psbt)
+        Ok(PayjoinProposal {
+            payjoin_psbt: self.payjoin_psbt,
+            owned_vouts: self.owned_vouts,
+            params: self.params,
+        })
+    }
+
+    pub fn finalize_proposal(
+        mut self,
+        wallet_process_psbt: impl Fn(&str) -> Result<WalletProcessPsbtResult, bitcoincore_rpc::Error>,
```

Comment:
This function signature has to be generic enough for downstream libraries without bitcoind to use it. It should be strongly typed. I imagine a signature like `impl Fn(&Psbt) -> Result<Psbt, Error>` would make the most sense. Perhaps the parameter should be owned.